lib/services/platform-environment-requirements.ts

```diff
@@ -0,0 +1,135 @@
+
+	private cliCommandToCloudCommandName: IStringDictionary = {
+		"build": "tns cloud build",
+		"run": "tns cloud run",
+		"deploy": "tns cloud deploy"
+	};
+
+	public async checkEnvironmentRequirements(platform: string): Promise<boolean> {
+		const canExecute = await this.$doctorService.canExecuteLocalBuild(platform);
+		if (!canExecute) {
+			if (!isInteractive()) {
+				this.fail("You are missing the nativescript-cloud extension and you will not be able to execute cloud builds. Your environment is not configured properly and you will not be able to execute local builds. To continue, choose one of the following options: " + EOL
+					+ "Run $ tns setup command to run the setup script to try to automatically configure your environment for local builds." + EOL
+					+ "Run $ tns cloud setup command to install the nativescript-cloud extension to configure your environment for cloud builds." + EOL
+					+ `Verify that your environment is configured according to the system requirements described at ${this.$staticConfig.SYS_REQUIREMENTS_LINK}`);
+			}
+
+			this.$logger.info("You are missing the nativescript-cloud extension and you will not be able to execute cloud builds. Your environment is not configured properly and you will not be able to execute local builds. " + EOL
+				+ "Select “Configure for Cloud Builds” to install the nativescript-cloud extension and automatically configure your environment for cloud builds." + EOL
+				+ "Select “Configure for Local Builds” to run the setup script and automatically configure your environment for local builds."
+				+ "Select “Configure for Both Local and Cloud Builds” to automatically configure your environment for both options."
+				+ "Select “Skip Step and Configure Manually” to disregard these options and install any required components manually.");
+
+			const selectedOption = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_MESSAGE, [
+				PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+				PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME,
+				PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME,
+				PlatformEnvironmentRequirements.BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME
+			]);
+
+			if (selectedOption === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
+				await this.processCloudBuilds(platform);
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME) {
+				await this.$doctorService.runSetupScript();
+
+				if (await this.$doctorService.canExecuteLocalBuild(platform)) {
+					return true;
+				}
+
+				const option = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_AFTER_SETUP_SCRIPT_MESSAGE, [
+					PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+					PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME
+				]);
+
+				if (option === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
```

Comment:
You can extract this piece of code
```ts
if (option === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
    await this.processCloudBuilds(platform);
}
```

into a method of this class and reuse it